openupgrade_scripts/scripts/mrp/18.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2025 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+add_columns = [
+    ("product.document", "attached_on_mrp", "char", "hidden"),
+]
+
+copy_columns = {
+    "mrp_production": [
+        ("location_dest_id", "location_final_id", "integer"),
+    ],
+}
+
+
+def mrp_workorder_sequence(env):
+    """
+    Precreate sequence field and fill with operation_id.sequence or 100
+    """
+    openupgrade.add_columns(env, [("mrp.workorder", "sequence", "integer", 100)])
+    env.cr.execute(
+        """
+        UPDATE mrp_workorder
+        SET sequence=COALESCE(mrp_routing_workcenter.sequence, mrp_workorder.sequence)
+        FROM mrp_routing_workcenter
+        WHERE mrp_workorder.operation_id=mrp_routing_workcenter.id
+        """
+    )
```

Comment:
```suggestion
    openupgrade.add_columns(env, [("mrp.workorder", "sequence", "integer")])
    env.cr.execute(
        """
        UPDATE mrp_workorder
        SET sequence=COALESCE(mrp_routing_workcenter.sequence, mrp_workorder.sequence)
        FROM mrp_routing_workcenter
        WHERE mrp_workorder.operation_id=mrp_routing_workcenter.id
        """
    )
    env.cr.execute(
        """
        SELECT COALESCE(max(sequence), 0) AS max_sequence
        FROM mrp_workorder
        WHERE sequence IS NOT NULL
        """
    )
    max_sequence = env.cr.fetchone()[0]
    openupgrade.logged_query(
        env.cr,
        f"""
        UPDATE mrp_workorder
        SET sequence = {max_sequence} + id::int
        WHERE sequence IS NULL""",
    )
```

Reply:
what's the point of this? In my version, work orders without operation_id are ordered by `leave_id, date_start, id` thanks to the default value you propose to delete. How is moving them after all orders with operation_id and enforcing an ordering by id only an improvement?